python/social_bot/envs/play_ground.py

```diff
@@ -235,14 +235,18 @@ def step(self, action):
             If with_language, it is a dictionary with key 'data' and 'sentence'
             If not with_language, it is a numpy.array or image for observation
         """
-        if self._with_language:
+        if False and self._with_language:
             sentence = action.get('sentence', None)
             if type(sentence) != str:
                 sentence = self._teacher.sequence_to_sentence(sentence)
             action_ctrl = action['control']
         else:
             sentence = ''
             action_ctrl = action
+        task = (self._teacher._current_task_group
+            ).get_current_task_non_generator()
+        if isinstance(task, GoalTask) and task._also_curriculum_control:
```

Comment:
task specific control_range might not be widely used for many tasks. I think we can override env.agent_control_range in GoalTask, or add a set_control_range to playground and use env.set_control_range() in GoalTask